### PR TITLE
fix: bump release type by dependent if deps.release is inherit

### PR DESCRIPTION
**Pull request by gavmck**
## Changes

When `deps.release` is set to `inherit`, but a package has changes, it overwrites the release type from the dependent package, even if the release type was greater (patch < minor < major).

This updates the `resolveRelesaseType` function so that if a dependent package has a larger version bump and `deps.release` is set to inherit, we'll take the inherited release type instead of the lesser local package one.

Test scenario: I had a major version bump in one of my packages that all other packages are dependent on and the dependent packages were only doing a minor update because they had local changes.

- [x] New code is covered by tests
- [ ] All the changes are mentioned in docs (readme.md)

### CODE REVIEW

File: lib/updateDeps.js [hunk 1]
```diff
@@ -163,23 +163,30 @@ const resolveReleaseType = (pkg, bumpStrategy = "override", releaseStrategy = "p
 	// NOTE This fn also updates pkg deps, so it must be invoked anyway.
 	const dependentReleaseType = getDependentRelease(pkg, bumpStrategy, releaseStrategy, ignore, prefix);
 
+	// Define release type for dependent package if any of its deps changes.
+	// `patch`, `minor`, `major` — strictly declare the release type that occurs when any dependency is updated.
+	// `inherit` — applies the "highest" release of updated deps to the package.
+	// For example, if any dep has a breaking change, `major` release will be applied to the all dependants up the chain.
+	// If we want to inherit the release type from the dependent package
+	const types = ['patch', 'minor', 'major'];
+	const depIndex = dependentReleaseType ? types.indexOf(dependentReleaseType) : 4;
```
Comment (antongolub) [inline lib/updateDeps.js:172]: Let's avoid using a magic number here, and assign the var to `types.length`.

File: lib/updateDeps.js [hunk 2]
```diff
@@ -163,23 +163,30 @@ const resolveReleaseType = (pkg, bumpStrategy = "override", releaseStrategy = "p
...
+	const pkgIndex = pkg._nextType? types.indexOf(pkg._nextType) : 4;
+
+	if (releaseStrategy === "inherit" && dependentReleaseType && depIndex >= pkgIndex) {
```
Comment (antongolub) [inline lib/updateDeps.js:175]: We check `dependentReleaseType` here, and the ~ `if (!dependentReleaseType)` below. It seems we could move this block down.

**Review by antongolub — CHANGES_REQUESTED** [review]

**Comment by antongolub** [comment]
Hey, @gavmck, 

Could you add a test for the fix?

File: lib/updateDeps.js (hunk 2 above)
Comment (gavmck) [reply lib/updateDeps.js:175]: I think it needs to return the `dependentReleaseType` before it tries to return `pkg._nextType` or we get unintended behaviour

**Comment by gavmck** [comment]
@antongolub Sorry, I know this has taken ages! 
Patched up feedback and added a test 👍

File: lib/updateDeps.js (hunk 2 above)
Comment (antongolub) [reply lib/updateDeps.js:175]: Let's add a minor tweak up: replace magic number `4` with `types.length`.

Comment (gavmck) [reply lib/updateDeps.js:175]: I think that's already been done?

**Review by antongolub — APPROVED** [review]
Seems good. Thanks for the contribution.
We will land this asap.

**Pull request merged**